03week/ticTacToe.js

```diff
@@ -23,32 +46,91 @@ function printBoard() {
   console.log('2 ' + board[2].join(' | '));
 }
 
-function horizontalWin() {
-  // Your code here
+const isValidInput = (row, column) => {
+  const validInputs = [0, 1, 2];
+  const isValid = validInputs.indexOf(row) > -1 || validInputs.indexOf(column) > -1
+  return isValid
+}
+
+const isOpenForPlay = (row, column) => {
+  return board[row][column] === ' '
+}
+
+const checksArrayForPlayerWin = (arr) => {
+  return arr.every((item) => item === playerTurn)
+}
+
+const checkMatrixForWin = (board) => {
+  let results = board.filter(checksArrayForPlayerWin);
+  return results.length > 0
+}
+
+const horizontalWin = () => {
+  return checkMatrixForWin(board);
 }
 
-function verticalWin() {
-  // Your code here
```

Comment:
excellent use of algorithms to check if every spot in the array is the same as the player turn.